staging/src/k8s.io/cloud-provider/node/helpers/taints.go

```diff
@@ -102,7 +97,17 @@ func PatchNodeTaints(c clientset.Interface, nodeName string, oldNode *v1.Node, n
 		return fmt.Errorf("failed to marshal new node %#v for node %q: %v", newNodeClone, nodeName, err)
 	}
 
-	patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, v1.Node{})
+	// Strip base diff node from RV to ensure that our Patch request will set RV to check for conflicts over .spec.taints.
+	// This is needed because .spec.taints does not specify patchMergeKey and patchStrategy and adding them is no longer an option for compatibility reasons.
+	// Using other Patch strategy works for adding new taints, however will not resolve problem with taint removal.
+	oldNodeNoRV := oldNode
+	oldNodeNoRV.ResourceVersion = ""
```

Comment:
yeah, same in the other file `pkg/controller/controller_utils.go`

Reply:
I was doing it, by Aldo pointed out that we don't need it because we are working on a single copy of the `oldNode`, from function which exits on return (see [taints.go](https://github.com/kubernetes/kubernetes/blob/32411016cc84d241558bc9738a525f5a8f7576d6/staging/src/k8s.io/cloud-provider/node/helpers/taints.go#L86))

Reply:
Moved back to deep copy, as it seems that performance gain is negligible and this might be a source of some funny bugs in the future.